test/connectors/parsers/v2/GetFinancialDetailsParserSpec.scala

```diff
@@ -44,6 +44,16 @@ class GetFinancialDetailsParserSpec extends AnyWordSpec with Matchers {
         | }
         |""".stripMargin
     ), headers = Map.empty)
+  val mockNotFoundHttpResponseWithBody: HttpResponse = HttpResponse.apply(status = Status.NOT_FOUND, json = Json.parse(
+    """
+      |{
+      | "documentDetails": [{
+      |   "summary": {
+      |     }
+      |   }]
+      | }
+      |""".stripMargin
```

Comment:
If it were a 404, I don't think the body would look like this, instead it would look like this:
```
        """
          |{
          | "failures":[
          |   {
          |     "code": "NO_DATA_FOUND",
          |     "reason": "This is a reason"
          |   }
          | ]
          |}
```